src/components/select/filterable-select/filterable-select.component.tsx

```diff
@@ -442,17 +444,17 @@ export const FilterableSelect = React.forwardRef(
     useEffect(() => {
       const textStartsWithFilter = textValue
         ?.toLowerCase()
-        .startsWith(filterText?.toLowerCase());
+        .startsWith(filterText?.trim().toLowerCase());
       const isTextboxActive = !disabled && !readOnly;
 
       if (
         isTextboxActive &&
         textboxRef &&
-        filterText?.length &&
-        textValue?.length > filterText?.length &&
+        filterText?.trimStart().length &&
```

Comment:
suggestion: you call this a few times here so it might be a but more readable to store it in a variable like you did above

``` 
const trimmed = filterText?.trimStart()